### DISCUSSION TITLE
Version 2.1.8

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#64] [Announcements]
**Improvements:**

- Added diagnostics
- Added the option to download integration or device diagnostics.

**Bugfixes:**

- Fixed the updating of the refresh interval that would not be set.
- Fixed the updating of the API key that would not be set.

<hr /><em>This discussion was created from the release <a href='https://github.com/rainepretorius/olarm-ha-integration/releases/tag/version2.1.8'>Version 2.1.8</a>.</em>

**Comment by rwoeke** [comment]
Hello

i installed 2.1.8. Please see log below

`This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/olarm_sensors/__init__.py:224 
Integration: Olarm Sensors (documentation, issues) 
First occurred: 06:36:13 (2 occurrences) 
Last logged: 06:38:09

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/olarm_sensors/__init__.py", line 224, in update_listener
    options[OLARM_DEVICES] = devices
                             ^^^^^^^
UnboundLocalError: cannot access local variable 'options' where it is not associated with a value
`
